platform/pulpcore/app/models/publication.py

```diff
@@ -0,0 +1,102 @@
+from django.db import models
+from django.core.files.storage import DefaultStorage, deconstructible
+
+from pulpcore.app.models import Model
+
+
+# Storage path calculator used by FileField.
+path_calculator = DefaultStorage().path_calculator
+
+
+class Publication(Model):
+    """
+    Fields:
+        created (models.DatetimeField): When the publication was created UTC.
+
+    Relations:
+        publisher (models.ForeignKey): The publisher that created the publication.
+    """
+
+    created = models.DateTimeField(blank=True, null=True, auto_now_add=True)
+
+    publisher = models.ForeignKey('Publisher', on_delete=models.CASCADE)
+
+
+class PublishedFile(Model):
+    """
+    A file included in Publication.
+
+    Fields:
+        relative_path (models.TextField): The (relative) path component of the published url.
+
+    Relations:
+        publication (models.ForeignKey): The publication in which the artifact is included.
+
+    """
+    relative_path = models.TextField(blank=False, null=False, max_length=255)
+
+    publication = models.ForeignKey(Publication, on_delete=models.CASCADE)
+
+    class Meta:
+        abstract = True
+
+
+class PublishedArtifact(PublishedFile):
+    """
+    An artifact that is part of a publication.
+
+    Relations:
+        artifact (models.ForeignKey): The referenced artifact.
+    """
+    artifact = models.ForeignKey('Artifact', blank=True, null=True, on_delete=models.CASCADE)
```

Comment:
This will get a second FK relation to a ``DeferredArtifact``/``RemoteArtifact`` or whatever we call it, right?